src/compiler/checker.ts

```diff
@@ -3050,6 +3051,11 @@ namespace ts {
             return undefined;
         }
 
+        function isAutoVariableInitializer(initializer: Expression) {
+            const expr = initializer && skipParentheses(initializer);
+            return !expr || expr.kind === SyntaxKind.NullKeyword || expr.kind === SyntaxKind.Identifier && getResolvedSymbol(<Identifier>expr) === undefinedSymbol;

```

Comment:
Yes, but they each need specialized strategies and I'll cover them in separate PRs.